### PR TITLE
Bump, major version

### PR DESCRIPTION
**Pull request by jessebeach**
Bump to v6.0.0

Depends on #255, #256

### CODE REVIEW

**Comment by evcohen** [comment]
what is breaking in this release/can we mark it in the changelog as breaking? if it's breaking, then let's include #250

**Comment by jessebeach** [comment]
It seemed break-ish, given the update to the configs. But let's put in #250 and make it break-ing.

**Comment by evcohen** [comment]
haha sounds like a plan 👍🏼

**Comment by AlmeroSteyn** [comment]
Currently, the `anchor-is-valid` rule is only included in the strict config. Should it, as it is breaking, be included in the recommended config, albeit with not all aspects activated? 

It is also overlapping the `href-no-hash` rule. The `invalidHref` aspect is incorporating the same check.

**Comment by jessebeach** [comment]
> It is also overlapping the href-no-hash rule. The invalidHref aspect is incorporating the same check.

Anyone opposed to deleting `href-no-hash` in favor of `anchor-is-valid`.

**Comment by ljharb** [comment]
As long as the test cases for href-no-hash are all indeed covered by anchor-is-valid, sgtm

**Comment by jessebeach** [comment]
> As long as the test cases for href-no-hash are all indeed covered by anchor-is-valid, sgtm

I'll verify.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,14 @@
+6.0.0 / 2017-06-05
+=================
+- [new] Add rule `anchor-is-valid`. See documentation for configuration options. Thanks @AlmeroSteyn.
+- [breaking] `href-no-hash` replaced with `anchor-is-valid` in the recommended and strict configs. `href-no-hash` will be deprecated in a future release.
```
Comment (ljharb) [inline CHANGELOG.md]: This text may need updating

Comment (jessebeach) [inline CHANGELOG.md]: Indeed. I had unstaged changes for this text. Changes are forthcoming. I've got some test fails from running the `href-no-hash` tests in `anchor-is-valid`. Just trying to work through them first.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "eslint-plugin-jsx-a11y",
-  "version": "5.0.3",
+  "version": "6.0.0",
```
Comment (evcohen) [inline package.json]: Can we also remove support for `eslint` v2 per #268

**Comment by AlmeroSteyn** [comment]
Are there any specific issues left you would like to get into this release? Would be happy to help with something.

**Comment by jessebeach** [comment]
@evcohen ready to merge!

**Comment by fearphage** [comment]
This caused facebookincubator/create-react-app#2631. There's nothing to be done here. The other issue is just locked so I can't comment there, but they need to be associated with each other.

**Comment by ljharb** [comment]
It didn’t “cause” it; its just that peer deps were incompatible.

**Pull request merged**